agents/cdu/fence_cdu.py

```diff
@@ -0,0 +1,143 @@
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, version 3 of the License.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#####
+##
+## The Following Agent Has Been Tested On:
+##
+##  Model                Firmware
+## +---------------------------------------------+
+## Sentry Switched CDU   6a
+## Sentry Switched CDU   7.1c <trenn@suse.de>
+##
+## @note: ssh has not being tested
+##
+#####
+
+import sys, re, pexpect, atexit
+sys.path.append("/usr/share/fence")
+from fencing import *
+from fencing import fail, EC_TIMED_OUT, run_command, frun
+
+def get_power_status(conn, options):
+    exp_result = 0
+    outlets = {}
+    try:
+        conn.send("STATUS\r\n")
+        conn.log_expect(options["--command-prompt"], int(options["--shell-timeout"]))
+        lines = conn.before.split("\n")
+        #    .A12     TowerA_Outlet12           On         Idle On  
+        #    .A12     test-01                   On         Idle On  
+        show_re = re.compile('(\.\w+)\s+(\w+|\w+\W\w+)\s+(On|Off)\s+(On|Idle On|Off|Wake On)')
+        for line in lines:
+            res = show_re.search(line)
+            if res != None:
+                outlets[res.group(2)] = (res.group(1), res.group(3))
+    except pexpect.EOF:
+        fail(EC_CONNECTION_LOST)
+    except pexpect.TIMEOUT:
+        fail(EC_TIMED_OUT)
+    try:
+        (_, status) = outlets[options["--plug"]]
+        return status.lower().strip()
+    except KeyError:
+        fail(EC_STATUS)
+
+def set_power_status(conn, options):
+        outlets = {}
+        action = { 'on' : "on", 'off': "off" }[options["--action"]]
+        try:
+            conn.send("LIST OUTLETS\r\n")
+            conn.log_expect(options["--command-prompt"], int(options["--shell-timeout"]))
+            lines = conn.before.split("\n")
+            #    .A12     TowerA_Outlet12
+            #    .A12     test-01
+            show_re = re.compile('(\.\w+)\s+(\w+|\w+\W\w+)\s+')
+            for line in lines:
+                res = show_re.search(line)
+                if res != None:
+                    outlets[res.group(2)] = (res.group(1))
+            conn.send(action + " " + outlets[options["--plug"]] + "\r\n")
+            conn.log_expect(options["--command-prompt"], int(options["--shell-timeout"]))
+        except pexpect.EOF:
+            fail(EC_CONNECTION_LOST)
+        except pexpect.TIMEOUT:
+            fail(EC_TIMED_OUT)
+
+def main():
+    device_opt = [  "help", "version", "agent", "quiet", "verbose", "debug",
```

Comment:
You should be able to remove lots of these (some are included by default, and others might be included by e.g. secure or similar (like inetX_only).

You can run `make xml-check` to see if you removed too many :)

Reply:
I removed most of it, including all ssh related options.
the old default -1 blowfish option showed that this won't work nowadays anymore anyway and it was mentioned untested at these times already.